### DISCUSSION TITLE
xDrip randomly stopped connected to G7 sensor with various errors

### DISCUSSION THREAD

**Discussion by BrianAllred** [NightscoutFoundation/xDrip#3685] [Bug reports]
xDrip version: aee9315-2024.09.19
Phone: Pixel 9 Pro Fold
Sensor: Dexcom G7
Pump: Tandem t:slim X2

I changed sensors Tuesday (9/24), started the sensor with my pump, and then connected to the sensor with xDrip. Everything was fine and I was getting readings on both my pump and phone.

Starting last night / early this morning, xDrip has stopped connecting to the sensor with various "Scanning Errors" and "Initializing Errors" and sensor status says "Mismatch - wait". 

To my knowledge, I didn't change anything in xDrip or my phone to cause this. I didn't install any system updates.

I tried uninstalling and reinstalling xDrip, which did not work. I continue to get accurate sensor readings through my pump, so I don't think it's the sensor.

**Comment by Navid200** [comment]
This is usually caused by not having disabled battery optimization correctly.
Can you show a screenshot?

**Reply by BrianAllred** [reply]
That might've might've caused the initial issue, but I made sure when I reinstalled that battery optimization was set to unrestricted.

I think somehow the Bluetooth pairing on the device got messed up. I unpaired all Dexcom devices (which there were only two, since this is a relatively new phone), then uninstalled and reinstalled the app again. After a few connection tries, it finally paired with the sensor and started getting readings again.

Note that I know Bluetooth is buggy sometimes, and my very first troubleshooting steps with sensor reading issues are always to toggle Bluetooth off and on, and then, if that doesn't work, reboot the phone.

So yeah, not entirely sure what fixed it, but it's good now. I guess I'll add "unpair the sensor in Bluetooth settings" to the troubleshooting list, lol.

**Reply by Navid200** [reply]
No, Bluetooth is not buggy.

You also need to set Legacy Bluetooth to unrestricted.
Have you done that?

**Reply by BrianAllred** [reply]
I'm not sure, where's that setting?

**Reply by Navid200** [reply]
Go to Android Settings -> Apps  -> See all apps

Now, tap on the ellipsis (3 dot) menu and tap on `Show System`.
Now, scroll down and you should find Bluetooth Legacy.

**Reply by BrianAllred** [reply]
Set it to Unrestricted, we'll see if that helps. Thanks!

**Reply by splitchard** [reply]
I just started having a similar issue on a Samsung Galaxy S22 phone. I do not see the Bluetooth Legacy app, only a Bluetooth app, a Bluetooth Agent app, and a Bluetooth MIDI Service app. How do I access the Legacy Bluetooth? Thanks!

**Reply by Navid200** [reply]
On some Pixel phones depending on the Android level, it may be called Legacy Bluetooth instead of Bluetooth.
If you find Bluetooth, that's what you need to disable battery optimization on completely the same as xDrip.

For Samsung, you can try enabling Special pairing workaround to see if it improves the situation or not.

It has also been advised to try enabling Trust Auto Connect again to see if it may improve the situation.

**Reply by splitchard** [reply]
Thank you for the response! I ended up updating to the latest 9/27 version of Xdrip and it started working (previously it was 9/17 version). As you pointed out, I also changed the Bluetooth battery setting from Optimized to Unrestricted. Hopefully this keeps it working!
